internal/store/hpa.go

```diff
@@ -143,6 +143,53 @@ var (
 				}
 			}),
 		},
+		{
+			Name: "kube_hpa_status_current_average_value",
+			Type: metric.Gauge,
+			Help: "Average metric value observed by the autoscaler.",
+			GenerateFunc: wrapHPAFunc(func(a *autoscaling.HorizontalPodAutoscaler) *metric.Family {
+				ms := make([]*metric.Metric, len(a.Status.CurrentMetrics))
+				for i, c := range a.Status.CurrentMetrics {
+					var value int64
+					if c.Type == autoscaling.ResourceMetricSourceType {
+						value, _ = c.Resource.CurrentAverageValue.AsInt64()
```

Comment:
I'd like to change the approach here. Can we `switch-case` statements instead of the `if-else-if`? Also, let's grab the `*resource.Quantity` type first from the switch-case block, Call `.AsInt64()` on the fetched `*resource.Quantity`,  check the 2nd return param of `AsInt64()` and then add it to the metric family only if the 2nd return param is true.